### PR TITLE
Add method for new search api for groups

### PR DESCRIPTION
**Pull request by rajendraBolla**

### CODE REVIEW

File: test/spec/api/search.spec.js [hunk 1]
```diff
@@ -54,6 +54,24 @@ describe('search api', function() {
...
+
+        describe('groups search', function() {
+            it('calls utilities.requestFun with correct request setup', function() {
+                search(apiOptions, utilitiesMock);
+                expect(utilitiesMock.requestFun).toHaveBeenCalledWith(
+                    jasmine.objectContaining({
+                        method: 'GET',
+                        resource: '/search/groups',
+                        headers: { 'Accept': MIME_TYPES.GROUP_SEARCH_RESULTS }
+                    })
```
Comment (customcommander) [inline test/spec/api/search.spec.js]: Missing `;`

File: lib/api/search.js [hunk 2]
```diff
@@ -47,6 +47,25 @@ module.exports = function search(options, utils) {
...
+         * @returns {promise}
+         */
+        groups: utils.requestFun({
+            authFlow: options.authFlow,
+            baseUrl: options.baseUrl,
+            method: 'GET',
+            resource: '/search/v2/groups',
+            headers: {
+              'Accept': MIME_TYPES.GROUP_SEARCH_RESULTS,
+              'Content-Type': MIME_TYPES.NULL
```
Comment (customcommander) [inline lib/api/search.js]: May I suggest `MIME_TYPES.NONE` instead?

Comment (achingbrain) [inline lib/api/search.js]: Or perhaps `MIME_TYPES.EMPTY`?

Comment (customcommander) [inline lib/api/search.js]: @rajendraBolla Would you mind changing to either NONE or EMPTY please?

Comment (RamIdeas) [inline lib/api/search.js]: We already have a NONE on the way to be set to 'application/none'

Comment (achingbrain) [inline lib/api/search.js]: It's still not clear why we are sending a `Content-Type` header at all.  The API docs for the corresponding service implies that it does not expect or require this header.

**Pull request merged**